### PR TITLE
Fix CSP unsafe-inline violation.

### LINKED ISSUE #84: ApplicationNavigation causes CSP inline style violation
# Bug Report

## Description
We are adding the `ApplicationNavigation` from `terra-application` and noticed the following CSP violation.  

Screen Shot 2020-09-09 at 3 42 11 AM

Upon digging deeper, It appears that `ApplicationStatusOverlayProvider`, included by the ApplicationNavigation [here](https://github.com/cerner/terra-application/blob/main/src/application-navigation/ApplicationNavigation.jsx#L12), requires a polyfill for IE10 that adds inline style dynamically without a nonce [here](https://github.com/cerner/terra-application/blob/9ac75245410caa2d6d59f87ead3650daab0e7033/src/application-status-overlay/ApplicationStatusOverlayProvider.jsx#L74-L76). 

Their doc  [here](https://wicg.github.io/inert/#strict-content-security-policy) suggests to add the following static style ourselves with the nonce and id

Screen Shot 2020-09-09 at 3 47 26 AM

but that seems odd since IE10 doesn't support CSP and other browsers don't need that polyfill. Apps that use the `ApplicationNavigation` with CSP is seeing this violation. 

## Steps to Reproduce

1.  Add `ApplicationNavigation` in an application
2. Turn on CSP without `unsafe-inline`  for `style-src` directive

## Additional Context / Screenshots
* it's coming from [here](https://github.com/WICG/inert/blob/master/src/inert.js#L702-L721)
Screen Shot 2020-09-09 at 12 19 52 PM

## Expected Behavior
* No CSP violation coming from `terra-application` without `unsafe-inline` for `style-src` directive

## Possible Solution
* Maybe we only require that polyfill for IE10 and not other browsers.

## Environment

* Component Name and Version: `ApplicationStatusOverlayProvider-1.34.0`
* Browser Name and Version: Chrome Version 85.0.4183.102 (Official Build) (64-bit)
* Node/npm Version: [e.g. Node 8/npm 5]  node 10
* Webpack Version: webpack ^4.41 
* Operating System and version (desktop or mobile): desktop

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

This PR provides [inert styles](https://github.com/WICG/inert/pull/148/files#diff-04c6e90faac2675aa89e2176d2eec7d8R101-R111) as global styles, and injects an empty link with id `inert-style` to head to avoid csp violation

**Note:** Currently the deployed PR will have a style tag in head. It will be removed once the [overlay PR-3166](https://github.com/cerner/terra-core/pull/3166) is merged

Closes #84 

### Deployment Link

https://terra-applic-.herokuapp.com/

### Testing

Tested on:
1. Chrome Version 85.0.4183.102 (Official Build) (64-bit) on Mac.
2. IE11
### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: src/application-status-overlay/ApplicationStatusOverlayProvider.jsx [hunk 1]
```diff
@@ -69,6 +69,15 @@ const ApplicationStatusOverlayProvider = ({ children, scrollRefCallback, ...cust
   const registeredVariant = registeredStatusOverlay[lastRegisteredStatusOverlayKey]?.variant;
 
   useEffect(() => {
+    const { head } = document;
+    if (!head.querySelector('link#inert-style')) {
+      if (head.querySelector('style#inert-style')) {
+        head.removeChild(head.querySelector('style#inert-style'));
+      }
```
Comment (mjhenkes) [inline src/application-status-overlay/ApplicationStatusOverlayProvider.jsx]: If the style block has already been added, the csp violation was caused by some other inert poly fill and we may as well leave it there, i think.

File: src/application-status-overlay/ApplicationStatusOverlayProvider.jsx [hunk 2]
```diff
@@ -69,6 +69,15 @@ const ApplicationStatusOverlayProvider = ({ children, scrollRefCallback, ...cust
   const registeredVariant = registeredStatusOverlay[lastRegisteredStatusOverlayKey]?.variant;
 
   useEffect(() => {
+    const { head } = document;
+    if (!head.querySelector('link#inert-style')) {
```
Comment (mjhenkes) [inline src/application-status-overlay/ApplicationStatusOverlayProvider.jsx]: lets wrap all this link injection under the if on line 82

File: src/application-status-overlay/ApplicationStatusOverlayProvider.jsx (hunk 1 above)
Comment (pranav300) [reply src/application-status-overlay/ApplicationStatusOverlayProvider.jsx]: Updated here https://github.com/cerner/terra-application/pull/89/commits/30513765b7611c3c5f527f56573286f521ce9c33.

File: src/application-status-overlay/ApplicationStatusOverlayProvider.jsx (hunk 2 above)
Comment (pranav300) [reply src/application-status-overlay/ApplicationStatusOverlayProvider.jsx]: Updated here https://github.com/cerner/terra-application/pull/89/commits/30513765b7611c3c5f527f56573286f521ce9c33.

Comment (pranav300) [reply src/application-status-overlay/ApplicationStatusOverlayProvider.jsx]: Updated here https://github.com/cerner/terra-application/pull/89/commits/30513765b7611c3c5f527f56573286f521ce9c33.

**Review by emilyrohrbough — APPROVED** [review]

**Review by saket2403 — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Review by tbiethman — CHANGES_REQUESTED** [review]
The status view has the same problem: https://github.com/cerner/terra-application/blob/main/src/application-status-overlay/ApplicationStatusOverlayProvider.jsx#L76

We could duplicate this change in that component as well, or you could take a look at what I'm thinking for a long term solution:
* https://github.com/cerner/terra-application/blob/application-v2-integration/src/application-base/private/initializeInert.js
* https://github.com/cerner/terra-application/blob/application-v2-integration/src/application-base/ApplicationBase.jsx#L19-L20

Theoretically such a change could go into terra-base today and cover all your open PRs to fix this. But we'd be including the inert-polyfill at all times, even if apps don't consume components that need it. The outstanding question there is whether or not we'd be initializing the MutationObserver polyfill more often in IE and degrading performance unnecessarily when inert is not needed.

Edit: Talked with @ryanthemanuel, we don't think this would cause an issue. @pranav300 how would you feel about making changes similar to my linked examples here in terra-application and closing out your other PRs? At this point most consumers (and even the consumer that logged the initial issue) are using terra-application and would benefit from a change here. 

With that in place, the other packages could remain as-is. They would still have their dynamic requires in place, but since we'd perform the initialization prior to their mounting, there would be no effect. Consumers not using terra-application and using a strict CSP could still run into issues, but that should be an infrequent occurrence both now and in the future.

File: src/application-base/ApplicationBase.jsx [hunk 3]
```diff
@@ -18,6 +18,9 @@ import { NavigationPromptCheckpoint } from '../navigation-prompt';
 import getBrowserLocale from './private/getBrowserLocale';
 import useTestOverrides from './private/useTestOverrides';
 
+import './private/initializeInert';
+import './private/inertStyles.scss';
```
Comment (mjhenkes) [inline src/application-base/ApplicationBase.jsx]: nit: Can you import the styles in `initializeInert` file?

File: src/application-base/private/initializeInert.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  // eslint-disable-next-line global-require
+  require('wicg-inert/dist/inert');
+}
+
+if (document.readyState === 'loading') {
+  /**
+   * The polyfill is executed after DOM content completes loading to ensure that the necessary elements
+   * are present prior to execution.
+   */
+  window.addEventListener('DOMContentLoaded', initializeInert);
```
Comment (mjhenkes) [inline src/application-base/private/initializeInert.js]: Lets attempt to remove this event listener in the initializeInert function.

File: src/application-base/ApplicationBase.jsx (hunk 3 above)
Comment (pranav300) [reply src/application-base/ApplicationBase.jsx]: updated here https://github.com/cerner/terra-application/pull/89/commits/b3a7c9601463e6bea305e88ebd392c3cbc9b820e

File: src/application-base/private/initializeInert.js (hunk 4 above)
Comment (pranav300) [reply src/application-base/private/initializeInert.js]: updated here https://github.com/cerner/terra-application/pull/89/commits/b3a7c9601463e6bea305e88ebd392c3cbc9b820e

**Review by mjhenkes — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,10 @@
 
 ## 1.35.0 - (September 22, 2020)
 
+* Changed
```
Comment (tbiethman) [inline CHANGELOG.md:5]: Can you move this into the Unreleased section? This probably just got merged weird with main.

Comment (pranav300) [reply CHANGELOG.md:5]: Updated here https://github.com/cerner/terra-application/pull/89/commits/3b6dae1c224f32327c8058d8bcc8b7280c108efa

**Review by tbiethman — APPROVED** [review]

**Pull request merged**